src/Microsoft.Azure.WebJobs.Host/Scale/ConcurrencyOptions.cs

```diff
@@ -0,0 +1,155 @@
+        /// and other plan info.
+        /// </remarks>
+        public long TotalAvailableMemoryBytes
+        { 
+            get
+            {
+                return _totalAvaliableMemoryBytes;
+            }
+            set
+            {
+                if (value <= 0 && value != -1)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(TotalAvailableMemoryBytes));
+                }
+
+                _totalAvaliableMemoryBytes = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the memory threshold dictating when memory based throttling
+        /// will kick in. Value should be between 0 and 1 indicating a memory percentage.
+        /// Set to -1 to disable memory based throttling.
+        /// </summary>
+        public float MemoryThreshold 
+        {
+            get
+            {
+                return _memoryThreshold;
+            }
+            set
+            {
+                if ((value <= 0 && value != -1) || (value >= 1))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(MemoryThreshold));
+                }
+
+                _memoryThreshold = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the CPU threshold dictating when cpu based throttling
+        /// will kick in. Value should be between 0 and 1 indicating a cpu percentage.
+        /// </summary>
+        public float CPUThreshold
+        {
+            get
+            {
+                return _cpuThreshold;
+            }
+            set
+            {
+                if (value <= 0 || value >= 1)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(CPUThreshold));
+                }
+
+                _cpuThreshold = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether persistent concurrency configuration snapshots
+        /// will be periodically written to persistent storage, to enable hosts to remember and
+        /// apply previously learned levels on startup.
+        /// </summary>
+        public bool SnapshotsEnabled { get; set; }
+
+        internal bool MemoryThrottleEnabled
+        {
+            get
+            {
+                return TotalAvailableMemoryBytes > 0 && MemoryThreshold > 0;
+            }
+        }
+
+        public string Format()
+        {
+            JObject options = new JObject
+            {
+                { nameof(DynamicConcurrencyEnabled), DynamicConcurrencyEnabled },
+                { nameof(MaximumFunctionConcurrency), MaximumFunctionConcurrency },
+                { nameof(TotalAvailableMemoryBytes), TotalAvailableMemoryBytes },
+                { nameof(MemoryThreshold), MemoryThreshold },
+                { nameof(CPUThreshold), CPUThreshold },
+                { nameof(SnapshotsEnabled), SnapshotsEnabled }
+            };
```

Comment:
This is the pattern we follow for all the other options types. The reason we do it this way is to avoid writing any sensitive data/values to logs, and also we might not want to log all values. So your comment makes sense and is technically correct, that's why we do it this way across the board.